Lesson_204V2/FullSolution/MainPage.xaml.cs

```diff
@@ -0,0 +1,196 @@
+            if (cdsVoltage < lowPotVoltage)
+            {
+                newState = eState.TooDark;
+            }
+
+            // And now check if it too high.
+            if (cdsVoltage > highPotVoltage)
+            {
+                newState = eState.TooBright;
+            }
+
+            // Use another method to determine what to do with the state.
+            await CheckForStateChange(newState);
+        }
+
+        private async Task CheckForStateChange(eState newState)
+        {
+            // Checks for state changes and does something when one is detected.
+            if (newState != CurrentState)
+            {
+                String whatToSay;
+
+                switch (newState)
+                {
+                    case eState.JustRight:
+                        {
+                            whatToSay = JustRightLightString;
+                        }
+                        break;
+
+                    case eState.TooBright:
+                        {
+                            whatToSay = HighLightString;
+                        }
+                        break;
+
+                    case eState.TooDark:
+                        {
+                            whatToSay = LowLightString;
+                        }
+                        break;
+
+                    default:
+                        {
+                            whatToSay = "unexpected value";
+                        }
+                        break;
+                }
+
+                // Use another method to wrap the speech synthesis functionality.
+                await TextToSpeech(whatToSay);
+
+                // Update the current state for next time.
+                CurrentState = newState;
+            }
+        }
+
+        private async Task TextToSpeech(String textToSpeak)
+        {
+            Debug.WriteLine(String.Format("MainPage::TextToSpeech {0}", textToSpeak));
+            // Because we are running somewhere other than the UI thread and we need to talk to a UI element (the media control)
+            // we need to use the dispatcher to move the calls to the right thread.
+            await Dispatcher.RunAsync(
+                Windows.UI.Core.CoreDispatcherPriority.High,
+                async () =>
+                {
+                    SpeechSynthesisStream synthesisStream;
+
+                    //creating a stream from the text which can be played using media element. This API converts text input into a stream.
+                    synthesisStream = await synthesizer.SynthesizeTextToStreamAsync(textToSpeak);
+
+                    // start this audio stream playing
+                    media.AutoPlay = true;
+                    media.SetSource(synthesisStream, synthesisStream.ContentType);
+                    media.Play();
+                }
+            );
+        }
+
+        // This will put our pin on the world map of makers
+        // Go to ENTER FINAL LINK HERE to view your pin
```

Comment:
"ENTER FINAL LINK HERE" should be http://ms-iot.github.io/content/en-US/win10/samples/BrightOrNot.htm